### ISSUE TITLE
Questions for run train.py

### ISSUE THREAD

**Issue by jiyoungAn** [JiahuiYu/generative_inpainting#41]
Hi, I am a beginner in tensorflow and I'm very interested in your paper.
I'll be very grateful if you could answer my question.

1. run only using CPU
I am trying to run train.py using CPU only.
So I changed NUM_GPUS to 0 (inpaint.yml).
However error related with reading gpu still has occurred and it stops after below sentence.
`32m[2018-05-23 19:02:54 @weights_viewer.py:60][0m Total size of trainable weights: 0G 10M 184K 136B (Assuming32-bit data type.)`
Are there any settings that need to be changed or Is it impossible to run only using cpu?

2. Adding train image in existing model
I'd like to add some training images(#800(shape 255,255,3)) in your place2 model.
1) downloading the place2 model you've built
2) locate the model into the model_logs file 
3) train
In this way, will my image be added to the existing model?

Regarding, jiyoungAn

**Comment by JiahuiYu** [comment]
1. Please remove [lines](https://github.com/JiahuiYu/generative_inpainting/blob/a2637d141727b2a7f9c3d2087e92691538eceb85/train.py#L34-L37) here. It is highly recommend to have GPUs to train model, otherwise the progress will be very slow.

2. Yes, you images with random sampled masks will be used to fine-tune existing model. Make sure you do have validation images to track the training progress, otherwise it may be easily overfitting on you 800 images.

**Comment by realczy** [comment]
Hi, I am able to run train.py using CPU only after removing the several code lines as you said.

But when I run `test.py`, it raises the `Error reading GPU information, set no GPU.`

What other changes should I make to run the test.py with your pretrained model still using CPU only?

Best Regards.

**Comment by JiahuiYu** [comment]
For testing, the similar lines should be removed in `test.py`.

**Comment by jiyoungAn** [comment]
thank you!!

**Comment by KangSH9776** [comment]
thank you very much!!

I have additional questions.

1.downloading the place2 model you've built
It's in the top question.

How and where to download place2 data?

Regarding, jiyoungAn

**Comment by jiyoungAn** [comment]
You can download it in "pretrained models" section here.
click dataset name.  :)

**Comment by KangSH9776** [comment]
Thank you for your answers.   
can I ask you one more question?
I'd like to do training about place2 data, not pretrained models.
How do I download place2 data  for  training?
Thank you very much for your time.

**Comment by JiahuiYu** [comment]
If you only have one GPU, starts from 0. `os.environ['CUDA_VISIBLE_DEVICES']='0'`

**Comment by KangSH9776** [comment]
Hello, I'm going to train 'place2 dataset', but I have a problem and leave a question.

I downloaded 'High-resolution images' as it was written. below 'Data of Places365-Standard '

The 'flist' file was downloaded from 'CelebA-HQ'.

I get the following 'cuda has no input image' error when executing 'python train.py '

I want to know how to do data when training 'place2'.

Thanks for reading

**Comment by JiahuiYu** [comment]
The path of files in flielist file maybe wrong. Please check. Please consider print the path inside data_from_fnames.py file.

**Comment by KangSH9776** [comment]
where is data_from_fnames.py file?

**Comment by knowtheworld** [comment]
> 1. Please remove [lines](https://github.com/JiahuiYu/generative_inpainting/blob/a2637d141727b2a7f9c3d2087e92691538eceb85/train.py#L34-L37) here. It is highly recommend to have GPUs to train model, otherwise the progress will be very slow.
> 2. Yes, you images with random sampled masks will be used to fine-tune existing model. Make sure you do have validation images to track the training progress, otherwise it may be easily overfitting on you 800 images.

It still stops after show the sentence:
[32m[2019-04-18 12:32:44 @weights_viewer.py:60][0m Total size of trainable wei
ghts: 0G 10M 184K 136B (Assuming32-bit data type.)
I have already removed the lines you said.How to fix this up?
